gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/TraversalRing.java

```diff
@@ -62,6 +62,10 @@ public void addTraversal(final Traversal.Admin<A, B> traversal) {
         this.traversals.add(traversal);
     }
 
+    public void setTraversal(final int index, final Traversal.Admin<A, B> traversal) {
```

Comment:
This is primarily a consistency thing. The speed you are talking about is negligible compared to the maintainability. We don't have any notion of "indexing" in TinkerPop traversals.